packages/commons/src/auth/jwt.ts

```diff
@@ -37,7 +45,12 @@ export const readAuthDataFromJwtToken = (
     } else {
       return {
         organizationId: token.data.organizationId,
-        userId: token.data.sub,
+        userId:
+          token.data.uid !== undefined
+            ? token.data.uid
+            : token.data.sub !== undefined
+            ? token.data.sub
+            : "",
```

Comment:
as far as I remember the `userId` should be taken only from the `uid` claim.
Did I forget a use case?

Reply:
In the `interops-common` repository inside the `package.scala` file I fond this line of code 
```
val uid: String                 = values.get(UID).filterNot(_.isBlank).orElse(values.get(SUB)).getOrElse("")
```
but reading the comments on top of the function
```
// TODO Kept for backward compatibility.
  //      To be removed as soon as all services have been migrated to the new authorize function
```
it seems something that should be deprecated.

Inside the `JWTReader.scala` file in the `bearerAsContext` function I found
```
    uid                 <- Try(Option(claims.getStringClaim(UID)).getOrElse(""))
    sub                 <- Try(Option(claims.getSubject).getOrElse(""))
```
so it seems that the context contains both fields.
I don't know wich field should be prioritized. I suspect that the correct field is `uid` and the `sub` is used only for backward compatibility.